File: app/src/main/java/com/example/cromai_desafio/ShowGraosCafe.java
package com.example.cromai_desafio;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;

public class ShowGraosCafe extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_show_graos_cafe);

        //Getting list in activity
        ListView List = findViewById(R.id.listCafe);
        //Getting extrar
        Intent intent=getIntent();
        String[] valueOfKey = intent.getStringArrayExtra("cafeArray");
        //Show array
        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, valueOfKey);//Set the adapter on my list
        List.setAdapter(adapter);
    }
}

File: app/src/main/java/com/example/cromai_desafio/ShowGraosCafe.kt
package com.example.cromai_desafio

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View

lateinit var cafes : Array<String>
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        cafes = arrayOf("Café Arábica", "Café Bourbon", "Café Kona", "Café Catuaí", "Café Acaiá", "Café Robusta", "Café Geisha")
    }

    fun changeScreen(view: View) {
        val intent = Intent(this, TiposGraosCafe::class.java)
            intent.putExtra("cafeArray", cafes)
        startActivity(intent)
        finish()
    }
}